llvm/lib/Target/X86/lab3/durandin_vladimir/X86VladimirMICounter.cpp

```diff
@@ -0,0 +1,73 @@
+#include "X86.h"
+#include "X86InstrInfo.h"
+#include "X86Subtarget.h"
+#include "llvm/CodeGen/MachineFunctionPass.h"
+#include "llvm/CodeGen/MachineInstrBuilder.h"
+
+#define MI_COUNTER_DESC "X86 Count number of machine instructions pass"
+#define MI_COUNTER_NAME "x86-vladimir-mi-counter"
+
+using namespace llvm;
+
+namespace {
+class X86VladimirMICounterPass : public MachineFunctionPass {
+public:
+  X86VladimirMICounterPass() : MachineFunctionPass(ID) {}
+
+  bool runOnMachineFunction(MachineFunction &MF) override;
+
+  static char ID;
+
+private:
+  StringRef getPassName() const override { return MI_COUNTER_DESC; }
+};
+} // end anonymous namespace
+
+char X86VladimirMICounterPass::ID = 0;
+
+bool X86VladimirMICounterPass::runOnMachineFunction(MachineFunction &MF) {
+  const TargetInstrInfo *TII = MF.getSubtarget().getInstrInfo();
+  DebugLoc DL3 = MF.front().begin()->getDebugLoc();
+
+  // Create a new virtual register
+  unsigned icReg = MF.getRegInfo().createVirtualRegister(&X86::GR64RegClass);
```

Comment:
Sorry for a long silence. I tried to find out the root cause of your problem.
Here is a possible solution:
```suggestion
  const TargetRegisterInfo *TRI = MF.getSubtarget().getRegisterInfo();

  DebugLoc DL3 = MF.front().begin()->getDebugLoc();

  // Create a new virtual register
  const TargetRegisterClass *PtrRC = TRI->getPointerRegClass(MF);

  unsigned icReg = MF.getRegInfo().createVirtualRegister(PtrRC);
```